sdk/storage/azure-storage-blob/azure/storage/blob/blob_client.py

```diff
@@ -133,6 +133,7 @@ def __init__(
         except AttributeError:
             raise ValueError("Blob URL must be a string.")
         parsed_url = urlparse(blob_url.rstrip('/'))
+        self.account_name = parsed_url.hostname.split('.')[0]
```

Comment:
This will not always work - a customer could be using a custom URL, it wont always be the storage account name

Reply:
See how this is handled in `generate_shared_access_signature` for reference